src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -98,6 +104,40 @@ public void removePerson(Person key) {
         persons.remove(key);
     }
 
+    //// assignment-level operations
+    /**
+     * Returns true if an assignment with the same identity as {@code assignment} exists in the person's assignment
+     * list.
+     */
+    public boolean hasAssignment(Name name, Assignment assignment) {
+        requireNonNull(assignment);
+        return persons.personWithSameName(name).getAssignments().contains(assignment);
+    }
+
+    /**
+     * Adds an assignment to the person's assignment list.
+     * The assignment must not already exist in the person's assignment list.
+     */
+    public void addAssignment(Name name, Assignment assignment) {
+        persons.personWithSameName(name).getAssignments().add(assignment);
+    }
+
+    /**
+     * Removes {@code key} from this {@code AddressBook's person assignment list}.
```

Comment:
I think "{@code AddressBook's person assignment list}" should be "{@code List<Assignment>}" or "{@code UniqueAssignmentList}" based on the implementation